libs/community/langchain_community/tools/firecrawl/tool.py

```diff
@@ -0,0 +1,127 @@
+import asyncio
+import logging
+import os
+from typing import List, Optional, Type
+
+from firecrawl import FirecrawlApp
+from langchain_core.callbacks import (
+    AsyncCallbackManagerForToolRun,
+    CallbackManagerForToolRun,
+)
+from langchain_core.tools import BaseTool
+from pydantic import BaseModel, Field
+
+# Configure logging
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+
+class ScrapeWebsiteInput(BaseModel):
+    base_url: str = Field(..., description="The base URL to scrape.")
+
+
+class FirecrawlScrapeWebsiteTool(BaseTool):
+    name: str = "scrape_website"
+    description: str = (
+        "A tool for scraping websites and extracting markdown content. "
+        "Provide the URL of the website to scrape and indicate if the scraping rate "
+        "should be limited. The parameters required are:\n"
+        "- 'base_url': The base URL of the website to scrape.\n"
+        "- 'limit_rate': A boolean specifying whether to limit the scraping rate."
+    )
+    args_schema: Type[BaseModel] = ScrapeWebsiteInput
+    return_direct: bool = True
+
+    api_key: str = Field(
+        default_factory=lambda: os.getenv("FIRECRAWL_API_KEY", ""),
+        description="API key for Firecrawl",
+    )
+    limit_rate: bool = Field(
```

Comment:
this is a confusing name -- could we rename to `rate_limit_enabled` and `FIRECRAWL_RATE_LIMIT_ENABLED`